### PR TITLE
Changes to updates left by staff

### PR DESCRIPTION
**Pull request by dracos**
Always show body name on normal staff updates even if marked anonymous, give Bexley a different message, and tidy up the docs related to this area.

Fixes https://github.com/mysociety/fixmystreet-commercial/issues/2002
Fixes https://github.com/mysociety/fixmystreet-commercial/issues/2003

~~Improvements to staff leaving report/updates, and display of body name. A staff user's update is always displayed as posted by the body user, but when making the update, the staff user is shown their own user's name and given the option of showing/not showing, which does not correspond with what then happens. So show the body name as the name that will be displayed so that's clear, always show a body name on a staff update (that isn't contribute as another user/anonymous user), and related change for Bexley.~~ Leaving this for now, because more complex and reports/updates don't behave the same, but I think ideally the update form for staff users should not show them their name/show name tickbox because what they select there does not matter. But with this PR it doesn't matter if they tick anon, and they can just leave the boxes alone, so this will probably do.

### CODE REVIEW

File: perllib/FixMyStreet/App/Controller/Report/New.pm [hunk 1]
```diff
@@ -1029,6 +1033,8 @@ sub process_report : Private {
         $report->anonymous(0);
     } elsif ($c->stash->{contributing_as_anonymous_user}) {
         $report->anonymous(1);
+    } elsif (!$c->stash->{contributing_as_another_user} && $c->user_exists && ($c->user->from_body || $c->user->is_superuser)) {
+        $report->anonymous(0);
     } else {
         $report->anonymous( $params{may_show_name} ? 0 : 1 );
     }
```
Comment (struan) [inline perllib/FixMyStreet/App/Controller/Report/New.pm:1040]: `contributing_as_another_user` is set in `process_user` which happens after `process_update` we never see the default condition for a staff user.

**Review by struan — CHANGES_REQUESTED** [review]
Few bugs in this :(. 

You still get "hide my name" links next to council names which seems contrary to the aim of this.

And the manual could do with an update in the "Creating reports on behalf of another user/ the council" section

File: perllib/FixMyStreet/App/Controller/Report/New.pm (hunk 1 above)
Comment (dracos) [reply perllib/FixMyStreet/App/Controller/Report/New.pm:1040]: It is set first in process_report, at the top of this function.

File: perllib/FixMyStreet/DB/Result/Comment.pm [hunk 2]
```diff
@@ -267,9 +267,13 @@ sub meta_line {
 
     my $meta = '';
 
-    if ($self->anonymous or !$self->name) {
+    my $contributed_as = $self->get_extra_metadata('contributed_as');
+    my $staff = $self->user->from_body || $self->get_extra_metadata('is_body_user') || $self->get_extra_metadata('is_superuser');
+    my $anon = $self->anonymous || !$self->name;
+
+    if ($anon && (!$staff || $contributed_as)) {
```
[the remark this answers is not in the document]
Comment (dracos) [reply perllib/FixMyStreet/DB/Result/Comment.pm:274]: Due to line 1032 of New.pm, contributed_as == body and anonymous == true can never happen together.

[the remark this answers is not in the document]
Comment (dracos) [reply perllib/FixMyStreet/DB/Result/Comment.pm:274]: Or equivalent line 311ish in Update.pm

[the remark this answers is not in the document]
Comment (struan) [reply perllib/FixMyStreet/DB/Result/Comment.pm:274]: Unless they click the "hide my name link" which is how I ended up with it :)

**Review by struan — APPROVED** [review]
👍

**Pull request merged**